draft-ietf-rats-eat.md

```diff
@@ -1198,104 +1150,40 @@ The string identifying the JSON-encoded token MUST be one of the following:
+mechanism as the enclosing token (e.g., it is signed by the same attestation key).
 
-This type of submodule consists of a digest made using a cryptographic hash of a Claims-Set.
-The Claims-Set is not included in the token.
-It is conveyed to the verifier outside of the token.
-The submodule containing the digest is called a detached digest.
-The separately conveyed Claims-Set is called a detached claims set.
+The encoding of a submodule Claims-Set MUST be the same as the encoding as the surrounding EAT.
 
-The input to the digest is exactly the byte-string wrapped encoded form of the Claims-Set for the submodule.
-That Claims-Set can include other submodules including nested tokens and detached digests.
+#### Nested-Token {#Nested-Token}
 
-The primary use for this is to facilitate the implementation of a small and secure attester, perhaps purely in hardware.
-This small, secure attester implements COSE signing and only a few claims, perhaps just UEID and hardware identification.
-It has inputs for digests of submodules, perhaps 32-byte hardware registers.
-Software running on the device constructs larger claim sets, perhaps very large, encodes them and digests them.
-The digests are written into the small secure attesters registers.
-The EAT produced by the small secure attester only contains the UEID, hardware identification and digests and is thus simple enough to be implemented in hardware.
-Probably, every data item in it is of fixed length.
+The Nested-Token type provides a means of representing claims from a submodule that has its own attesting environment,
+i.e., it has keys distinct from the attester producing the surrounding token. Claims are represented in a signed EAT token. Inclusion of a signed
+EAT as a claim cryptographically binds the EAT to the surrounding token.
+If it was conveyed in parallel with the surrounding token, there would be no such binding and attackers could substitute a good attestation from another device for the attestation of an errant subsystem.
 
-The integrity protection for the larger claims sets will not be as secure as those originating in hardware block, but the key material and hardware-based claims will be.
-It is possible for the hardware to enforce hardware access control (memory protection)  on the digest registers so that some of the larger claims can be more secure.
-For example, one register may be writable only by the TEE, so the detached claims from the TEE will have TEE-level security.
+A nested token need not use the same encoding as the enclosing token.
+This enables composite devices to be built without regards to the encoding used by components.
+Thus, a CBOR-encoded EAT can have a JSON-encoded EAT as a nested token and vice versa.
 
-The data type for this type of submodule MUST be an array
-It contains two data items, a hash algorithm identifier and a byte string containing the digest.
+#### Detached Submodule Digest
```

Comment:
I wonder about re ordering so that detached comes after submodule claims-set.  Both use the same attesting environment as the surrounding token. I think that is an important point (that the previous text didn't make either). It will be easier to make this point with the re ordering.

Reply:
Nested-Token was moved to come after Detached Submodule Digest and re-ordered the bullets and CDDL in the front part of the section. Added a sentence re: attesting environment to the front part as well.